### PR TITLE
Generalize model selection a bit.

### PR DESCRIPTION
**Pull request by codeviking**
This factors things a bit so that model selection is captured
in the `useModels` hook rather than the task code itself. The
motivation being to reduce the amount of boilerplate that exists
in each demo.

To do this I modified `useModels` to return a class I call a
`ModelList`, which is just a wrapper for a list of models with
a reference to the selected one. This way demos can simply refer
to `list.selected`, rather than managing that value themselves.

The function also returns a second parameter, which is a function
that updates the selected model.

I'm still getting used to using React hooks, but this feels pretty
nice. Curious to see what y'all think.

### CODE REVIEW

File: ui/src/demos/reading-comprehension/Main.tsx [hunk 1]
```diff
@@ -100,13 +87,13 @@ export const Main = () => {
     const handleFormSubmit = async (i: Input) => {
         // TODO: Show some form of loading.
         // TODO: Implement better error handling (show something nice to the user).
```
Comment (jonborchardt) [inline ui/src/demos/reading-comprehension/Main.tsx:89]: I believe this whole method could live to model...

**Review by jonborchardt — APPROVED** [review]
Very nice

File: ui/src/demos/reading-comprehension/Main.tsx (hunk 1 above)
Comment (codeviking) [reply ui/src/demos/reading-comprehension/Main.tsx:89]: Yea, I think you're right. I put that on my list of things to explore today.

File: ui/src/lib/Model.ts [hunk 2]
```diff
@@ -22,12 +22,49 @@ export class Model<I, O> {
     }
 }
 
-export function useModels<I, O>(...ids: string[]): Model<I, O>[] | undefined {
-    const [models, setModels] = React.useState<Model<I, O>[]>();
+class ModelNotFoundError extends Error {
```
Comment (jonborchardt) [inline ui/src/lib/Model.ts:25]: nice

**Pull request merged**